src/inference/src/cache_manager.hpp

```diff
@@ -99,16 +99,32 @@ class FileStorageCacheManager final : public ICacheManager {
 
 private:
     void write_cache_entry(const std::string& id, StreamWriter writer) override {
+        // Fix the bug caused by pugixml, which may return unexpected results if the locale is different from "C".
+        auto original_plocal = setlocale(LC_NUMERIC, nullptr);
+        if (std::strcmp(original_plocal, "C") != 0) {
+            setlocale(LC_ALL, "C");
+        }
         std::ofstream stream(getBlobFile(id), std::ios_base::binary | std::ofstream::out);
         writer(stream);
+        if (std::strcmp(original_plocal, "C") != 0) {
+            setlocale(LC_ALL, original_plocal);
+        }
     }
 
     void read_cache_entry(const std::string& id, StreamReader reader) override {
+        // Fix the bug caused by pugixml, which may return unexpected results if the locale is different from "C".
+        auto original_plocal = setlocale(LC_NUMERIC, nullptr);
```

Comment:
Something to be careful about here -- the char* that setlocale is returning here may be a reference to the current, globally set locale. So, by calling setlocale below, it be changing the string that 'original_plocal' is pointing to. 

Just a disclaimer... I read something like that from some stack overflow thread -- I don't know if it's true :smile: